score-data-format/src/main/java/io/cloudslang/content/json/services/JsonService.java

```diff
@@ -0,0 +1,154 @@
+import com.jayway.jsonpath.internal.spi.json.GsonJsonProvider;
+import com.jayway.jsonpath.spi.json.JsonProvider;
+import io.cloudslang.content.json.exceptions.RemoveEmptyElementException;
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import net.minidev.json.JSONStyle;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Created by Folea Ilie Cristian on 2/3/2016.
+ */
+public class JsonService {
+
+    public String removeEmptyElementsJson(String json) throws RemoveEmptyElementException {
+        String normalizedJson = removeBlanks(json);
+
+        char wrappingQuote = retrieveWrappingQuoteTypeOfJsonMemberNames(normalizedJson);
+
+        Map<String, Object> jsonMap;
+
+        try {
+
+            parseJsonForInconsistencies(normalizedJson);
+            jsonMap = JsonPath.read(normalizedJson, "$");
+        } catch (com.jayway.jsonpath.InvalidJsonException | com.google.gson.JsonSyntaxException ije) {
+            throw new RemoveEmptyElementException(ije);
+        }
+
+        removeEmptyElementsFromMap(jsonMap);
+        return generateResultingJsonString(wrappingQuote, jsonMap);
+    }
+
+    private String generateResultingJsonString(char wrappingQuote, Map<String, Object> jsonMap) {
+        JSONObject jsonObject = new JSONObject();
+
+        for(String key:jsonMap.keySet()){
+             jsonObject.put(key,jsonMap.get(key));
+        }
+
+        String newJson = jsonObject.toJSONString(JSONStyle.LT_COMPRESS);
+
+        if (newJson.charAt(1) != wrappingQuote) {
+            return replaceUnescapedOccurrencesOfCharacterInText(newJson, newJson.charAt(1), wrappingQuote);
+        }
+
+        return newJson;
+    }
+
+
+    private void parseJsonForInconsistencies(String normalizedJson) {
+        JsonProvider provider = new GsonJsonProvider();
+        Configuration configuration =  Configuration.builder().jsonProvider(provider).build();
+        JsonPath.parse(normalizedJson, configuration);       //throws an exception at runtime if the json is malformed
+    }
+
+    /**
+     * Returns the quote character used for specifying json member names and String values of json members
+     * @param jsonString the source json from which to extract the wrapping quote
+     * @return either one of the characters ' (single quote)or " (double quote)
+     */
+    private char retrieveWrappingQuoteTypeOfJsonMemberNames(String jsonString) {
+        char quote = '\"';   //  the default quote character used to specify json member names and string value according to the json specification
+        for (char c : jsonString.toCharArray()) {
+            if (c == '\'' || c == '\"') {
+                quote = c;
+                break;
+            }
+        }
+        return quote;
+    }
+
+    private void removeEmptyElementsFromMap(Map<String, Object> json) {
+        LinkedHashMap<String, Object> copyJson = new LinkedHashMap<String, Object>(json);
+        for (Object element : copyJson.values()) {
```

Comment:
maybe you could use an iterator over the entrySet here to iterate and remove elements without needing a copy of the map